pages/intro.js

```diff
@@ -0,0 +1,186 @@
+                // if the textarea is the header (not resizable)
+                header = textarea.value
+                updateObj.WebsiteData[type]['header'] = header
+            } else {
+                // if text area is content (resizable)
+                content = textarea.value
+                updateObj.WebsiteData[type]['content'] = content
+            }
+        })
+        // TODO need to substitute the value at .doc(ID) with current website
+        await db.collection('Hackathons').doc('IntroTestHackathon')
+            .update(updateObj)
+            .then(() => {
+                setWebsiteData({
+                    ...websiteData,
+                    [type]: {
+                        ...websiteData[type],
+                        header,
+                        content,
+                        time
+                    }
+                })
+            })
+            .catch(() => {
+                alert('An error occured while trying to update the text section')
+            })
+        setIsEditingObj({
+            ...isEditingObj,
+            [type]: false
+        })
+    }
+
+    function Content(props) {
+        const { header, content, isEditing, type } = props
+        return (
+            <React.Fragment>
+                {
+                    isEditing ?
+                        <div style={{padding: '0px 40px 37px 40px'}}>
+                            <StyledHeader>Header</StyledHeader>
+                            <TextBox otherClassNames={type} defaultValue={header} />
+                            <StyledHeader>Body</StyledHeader>
+                            <TextBox otherClassNames={type} defaultValue={content} resize={true}/>
+                            <div style={{ marginTop: '27px', display: 'flex', float: 'right' }}>
+                                <StyledCancel onClick={() => handleCancel(type)}>
+                                    <p style={{ borderBottom: `2px solid ${COLOR.BLACK}`, margin: '0px' }}>
+                                        Cancel
+                                    </p>
+                                </StyledCancel>
+                                <Button onClick={() => handleSave(type)}>Save</Button>
+                            </div>
+                        </div>
+                        :
+                        <React.Fragment>
+                            <HeaderText>{header}</HeaderText>
+                            <ContentText>{content}</ContentText>
+                        </React.Fragment>
+                }
+            </React.Fragment>
+        )
+    }
+
+    // map over every text section in a hackathon's WebsiteData and adds the section name to isEditingObj state 
+    return (
+        <React.Fragment>
+        {Object.keys(websiteData).map((type) => {
+            if (isEditingObj[type] === undefined) {
+                setIsEditingObj({
+                    ...isEditingObj,
+                    [type]: false
+                })
+            }
```

Comment:
I think you should put this code in the same useEffect I mentioned about for getting the actual hackathon data, that way it won't run on every rerender.